scss/components/_title-bar.scss

```diff
@@ -42,7 +42,7 @@ $titlebar-icon-spacing: 0.25rem !default;
 
     @if $global-flexbox {
       display: flex;
-      justify-content: space-between;
+      justify-content: flex-start;
```

Comment:
I don't see why this should be start (left align)

By default I think it should be space-between is you had 2 sides like:
http://foundation.zurb.com/sites/docs/off-canvas.html#combining-with-title-bar